ext/datadog_profiling_native_extension/heap_recorder.c

```diff
@@ -635,8 +598,10 @@ static int st_object_records_iterate(DDTRACE_UNUSED st_data_t key, st_data_t val
     locations[i] = (ddog_prof_Location) {
       .mapping = {.filename = DDOG_CHARSLICE_C(""), .build_id = DDOG_CHARSLICE_C("")},
       .function = {
-        .name = {.ptr = frame->name, .len = strlen(frame->name)},
-        .filename = {.ptr = frame->filename, .len = strlen(frame->filename)},
+        .name = DDOG_CHARSLICE_C(""),
+        .name_id = frame->name,
+        .filename = DDOG_CHARSLICE_C(""),
+        .filename_id = frame->filename,
```

Comment:
I guess we'll see how this evolves on the libdatadog but I still think it's really weird/bizarre/meh that libdatadog doesn't allow us to just use NULL for strings in situations where they can be empty, as this looks a lot more awkward because of it.